storage/oasis/nodeapi/universal_runtime.go

```diff
@@ -136,3 +140,16 @@ func (rc *UniversalRuntimeApiLite) GetEventsRaw(ctx context.Context, round uint6
 func (rc *UniversalRuntimeApiLite) EVMSimulateCall(ctx context.Context, round uint64, gasPrice []byte, gasLimit uint64, caller []byte, address []byte, value []byte, data []byte) ([]byte, error) {
 	return evm.NewV1(rc.sdkClient).SimulateCall(ctx, round, gasPrice, gasLimit, caller, address, value, data)
 }
+
+func (rc *UniversalRuntimeApiLite) GetNativeBalance(ctx context.Context, round uint64, addr Address) (*common.BigInt, error) {
+	balances, err := accounts.NewV1(rc.sdkClient).Balances(ctx, round, sdkTypes.Address(addr))
+	if err != nil {
+		return nil, err
+	}
+	nativeBalance, ok := balances.Balances[sdkTypes.NativeDenomination]
```

Comment:
oh right, because we have support for multiple denominations in the accounts module :neutral_face: 

need to ask some questions about this

1. architecturally what would it be like to support other denominations? more reserved addresses and token type integers?
2. will the indexer team ever put pressure on paratime developers including OPF never to use multiple denominations?
3. does the indexer understand that the native denomination of all paratimes and consensus staking token are the same (modulo decimals changes)? this is incidentally true of the networks and paratimes we currently support, but paratimes can be configured otherwise

Reply:
I'm working with the assumption that no EVM (!) runtime will ever actually introduce multiple runtime-levle tokens. Or that if they do, then the non-native runtime-level tokens (because even among the denominations specified by the `accounts` module, one is first among equals!) will be trackable via runtime events and won't require EVM querying.

If this assumption is proven wrong, these are good questions and I think we'd have to rethink/redesign things a bit at that point.
Re 1, yeah it's an option as a hack to support one or two known hardcoded denominations, but it gets hairy fast.
Re 2, I don't think that's our place, we have to adapt to the chain, not the other way around. (We did discuss the node adding a handy API extension here or there at our behest, but that's different.)
Re 3, I don't think we make that assumption anywhere.

Reply:
thanks for the analysis. you're right, the evm does only touch one accounts module denomination at a time. https://github.com/oasisprotocol/emerald-paratime/blob/v10.0.0/src/lib.rs#L69 that's configured here, which for the current paratimes we support, it's the native token.